lib/inc/internal/util/aix/odm.hpp

```diff
@@ -0,0 +1,263 @@
+#include <odmi.h>
+
+#include <cstdint>
+#include <memory>
+#include <string>
+
+namespace facter { namespace util { namespace aix {
+
+    // This is a singleton class that handles initialization and
+    // termination of the ODM library. As long as any other class holds a
+    // reference to this one, ODM will remain initialized.
+    //
+    // This class also provides static helpers for interacting with the ODM
+    class odm {
+    public:
+         typedef std::shared_ptr<odm> ptr;
+         static ptr get() {
+             static std::weak_ptr<odm> self;
```

Comment:
It's not thread-safe in this form, because I don't have std::mutex on AIX and didn't want to mess with low-level locking primitives. With a bit of cleanup we could probably turn it into a mixin (ala `enable_shared_from_this`) - something like `class my_thing : public lazy_shared_singleton<my_thing>`